### PR TITLE
[Build] Improve Windows build script

### PR DESCRIPTION
**Pull request by PENGUINLIONG**
- Support Python package userspace installtion;
- Support downloading prebuilt LLVM and Clang if absent;
- Powershell functions only, no depedency on environmental software, should work with any Windows 10+.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -1,3 +1,126 @@
...
+}
+
+# Install in userspace?
+$BuildExpr = "python setup.py develop";
+if ($UserSpace) {
+    Write-Host "Taichi Python package will be installed in user-space."
+    $BuildExpr += " --user"
+}
+
+Write-Host
```
Comment (qiao-bo) [inline build.ps1:120]: is this line intended?

**Review by qiao-bo — APPROVED** [review]
Thanks for cleaning-up this script!

File: build.ps1 (hunk 1 above)
Comment (PENGUINLIONG) [reply build.ps1:120]: Yes, to separate the log a bit from those emitted by the actual building script. 😮

**Pull request merged**